src/middlewares/authenticate.ts

```diff
@@ -0,0 +1,16 @@
+import { NextFunction, Request, Response } from "express";
+import jwt from "jsonwebtoken";
+
+const authenticate =async (req: Request, res: Response, next: NextFunction) => {
+    console.log("in authenticate middleware");
+    const accesTokenByReactAsHeader=req.headers.authorization?.split(" ")[1];
+    try{
+    const pass=await jwt.verify(accesTokenByReactAsHeader as string,process.env.JWT_SECRET as string)
+    if(pass)
+    next();
+    }
+    catch{
+        return res.send('not authorized');
```

Comment:
use CustomeError middleware. If you have not created one. use `
res.status(401);
throw new Error("Not Authorized");
`

Reply:
ok